### PR TITLE
Big queued price deviation

### PR DESCRIPTION
**Pull request by mauricio1802**
This PR adds a handler on the Oracle Security Agent for checking Oracles that enqueue prices with big deviations

### CODE REVIEW

File: Maker-Agents-Suit-Module/OSMAgent/src/big.queued.price.deviation.spec.ts [hunk 1]
```diff
@@ -0,0 +1,296 @@
...
+  it("should returns empty findings if the correct event wasn't logged", async () => {
+    handleTransaction = provideBigQueuedPriceDeviationHandler(
+      CONTRACT_ADDRESSES
+    );
+
+    const txEvent: TransactionEvent = new TestTransactionEvent()
+      .addTrace({
+        from: CONTRACT_ADDRESSES[0],
+        input: PEEK_FUNCTION_SELECTOR,
+        output: web3.eth.abi.encodeParameters(["uint128", "bool"], [108, true]),
```
Comment (e1Ru1o) [inline Maker-Agents-Suit-Module/OSMAgent/src/big.queued.price.deviation.spec.ts:164]: Same here as above, I think that the problem is that you wrote tests for different `lenght` in the results of next and current, being consistent with the current implementation

File: Maker-Agents-Suit-Module/OSMAgent/src/big.queued.price.deviation.ts [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+  contractAddress: string,
+  txEvent: TransactionEvent
+): Finding | null => {
+  const currentValues: bigint[] = getCurrentValues(contractAddress, txEvent);
+  const nextValues: bigint[] = getNextValuesForOSM(
+    contractAddress,
+    txEvent.traces
+  );
+
+  const lessLenght: number = Math.min(currentValues.length, nextValues.length);
```
[the remark this answers is not in the document]
Comment (mauricio1802) [reply Maker-Agents-Suit-Module/OSMAgent/src/big.queued.price.deviation.ts:94]: You are right, I am going to remove that line.

**Review by CeliktepeMurat — APPROVED** [review]

File: Maker-Agents-Suit-Module/OSMAgent/src/big.queued.price.deviation.ts [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+  for (let i = 0; i < lessLenght; i++) {
```
Comment (CeliktepeMurat) [inline Maker-Agents-Suit-Module/OSMAgent/src/big.queued.price.deviation.ts:95]: May you elaborate this length topic. I'm just confused about lessLength and loop over it etc.

Comment (mauricio1802) [reply Maker-Agents-Suit-Module/OSMAgent/src/big.queued.price.deviation.ts:95]: I've already removed it. I thought I could have multiple lengths and in that case I wanted to iterate until one of the arrays were consumed. But both arrays must have the same length.

**Review by e1Ru1o — APPROVED** [review]

**Review by rachit2501 — APPROVED** [review]

**Pull request merged**